lib/aca_entities/medicaid/ios/functions/ssp_asset__c_builder.rb

```diff
@@ -1,33 +1,45 @@
 # frozen_string_literal: true
 
+require 'aca_entities/medicaid/ios/types'
+
 module AcaEntities
   module Medicaid
     module Ios
       module Functions
         # build SSP_Asset__c for IOS transform
         class SspAssetCBuilder
-          def call(cache)
-            @memoized_data = cache
-            # TO DO
-            # loop through incomes and deductions
-            # build hash of field mappings
-            # return array of transformed SSP_Asset__c hashes
-            [] # mocked return array for initial spec
 
-            # application = @memoized_data.resolve('family.magi_medicaid_applications').item
-            # applicants = application[:applicants]
-            # incomes = applicants.map {|applicant| applicant[:incomes]}.flatten
-            # deductions = applicants.map {|applicant| applicant[:deductions]}.flatten
-
-            # asset_incomes = incomes.map do |income|
-            #   {}
-            # end
+          def call(cache)
+            application = cache.resolve('family.magi_medicaid_applications').item
+            applicants = application[:applicants]
+            incomes = applicants.map {|applicant| applicant[:incomes]}.flatten
+            deductions = applicants.map {|applicant| applicant[:deductions]}.flatten
 
-            # asset_deductions = deductions.each do |deduction|
-            #   {}
-            # end
+            asset_incomes = incomes.map do |income|
+              {
+                # 'Id' => generate_hash,
+                'IncomeTypeCode__c' => AcaEntities::Medicaid::Ios::Types::INCOME_TYPE_MAP[income&.dig(:kind)&.upcase],
+                # 'IncomeSubtypeCode__c' => income&.dig(:subtype)&.upcase,
+                'StartDate__c' => income&.dig(:start_on),
+                'TotalGrossAmount__c' => income&.dig(:amount),
+                'IncomePayFrequency__c' => AcaEntities::Medicaid::Ios::Types::FREQUENCY_MAP[income&.dig(:frequency_kind)&.upcase],
+                'EndDate__c' => income&.dig(:end_on),
+                'EmployerName__c' => income&.dig(:employer, :employer_name),
+                'PrimaryPhoneNumber__c' => income&.dig(:employer, :employer_phone, :full_phone_number)
+              }
+            end
 
-            # asset_incomes + asset_deductions
+            asset_deductions = deductions.map do |deduction|
+              {
+                # 'Id' => generate_hash,
```

Comment:
Do we have a question around the Id field that we are waiting on a response for?  I'm wondering why it's commented out and what the purpose of generate_hash is, since the contract is expecting a string for this field.